### PR TITLE
LT-1029 Templates: code review comments

### PR DESCRIPTION
**Pull request by BenAndrew-Kainos**

### CODE REVIEW

File: ui_automation_tests/features/case_view_details.feature [hunk 1]
```diff
@@ -13,3 +13,11 @@ Feature: I want to view the case details of a case
...
     And I see a third party
     And I see a consignee
+
+  @LT_982_exporter_edited_case_anchor
+  Scenario: Gov user can see exporter has made changes to case
+    Given I create application or application has been previously created
+    And I sign in to SSO or am signed into SSO
+    When the exporter user has edited the case
+    And I go to application previously created
+    Then I see that changes have been made to the case
```
Comment (craiga) [inline ui_automation_tests/features/case_view_details.feature]: What is this?

File: cases/views/main.py [hunk 2]
```diff
@@ -18,7 +18,7 @@
 from conf.settings import AWS_STORAGE_BUCKET_NAME
 from core.builtins.custom_tags import get_string
 from core.helpers import convert_dict_to_query_params
-from core.services import get_user_permissions, get_statuses
+from core.services import get_user_permissions, get_statuses, get_user_case_notification
```
[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply cases/views/main.py]: This is from merging master

File: picklists/views.py [hunk 3]
```diff
@@ -89,23 +86,6 @@ def get(self, request, **kwargs):
...
-                # Template is valid! :)
-            except TemplateSyntaxError as err:
-                # Template is invalid! :(
-                return form_page(
-                    request,
-                    self.form,
-                    data=request.POST,
-                    # err.args contains error messages from template engine.
-                    errors={"text": err.args},
-                )
```
[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply picklists/views.py]: This was only being executed if the picklist type is paragraph but no picklist_type field is ever passed.

[the remark this answers is not in the document]
Comment (BenAndrew-Kainos) [reply picklists/views.py]: All fixed with added variable checking :)

File: letter_templates/helpers.py [hunk 4]
```diff
@@ -5,19 +5,30 @@
...
+
+
+def template_engine_factory(allow_missing_variables=False):
     """
     Create a template engine configured for use with letter templates.
     """
+    string_if_invalid = '{{ %s }}' if allow_missing_variables else InvalidVarException()
     return Engine(
-        string_if_invalid='{{ %s }}',
+        string_if_invalid=string_if_invalid,
```
Comment (craiga) [inline letter_templates/helpers.py]: Cool; does this throw an `InvalidVarException` if a variable is missing?

Could you include a link to documentation for that here?

File: picklists/forms.py [hunk 5]
```diff
@@ -54,6 +54,7 @@ def edit_picklist_item_form(picklist_item):
                 questions=[
                     _name,
                     conditional(picklist_item['type']['key'] == 'letter_paragraph', _paragraph, _text),
+                    HiddenField('type', picklist_item['type']['key'])
```
Comment (craiga) [inline picklists/forms.py]: Ah, so `type` was only included in the form when creating a picklist item? If so, I missed that. Thanks for picking it up.

File: letter_templates/helpers.py (hunk 4 above)
Comment (craiga) [reply letter_templates/helpers.py]: Oh wait, `InvalidVarException` is a clever hack to trick the Django Template Engine into doing this … cool.

We should document what's going on in a comment though.

**Pull request merged**